db/version_set.cc

```diff
@@ -4594,7 +4594,11 @@ Status VersionSet::Recover(
       Version* v = new Version(cfd, this, file_options_,
                                *cfd->GetLatestMutableCFOptions(),
                                current_version_number_++);
-      builder->SaveTo(v->storage_info());
+      s = builder->SaveTo(v->storage_info());
+      if (!s.ok()) {
```

Comment:
Maybe try to reduce exit point of this function to simplify memory reclamation, given that we are not using smart pointers for `Version` objects, etc.

Reply:
nvm, I think we can live with a `delete v` here in this PR.